apps/dashboard/app/javascript/packs/icon_picker.js

```diff
@@ -2,108 +2,120 @@
 
 import ALL_ICONS from './icons';
 
-const ICON_SHOW_ID = "product_icon"
-const ICON_SELECT_ID = "product_icon_select"
-
-// simple boolean for whether there are currently any hidden icons
-let hiddenIcons = false;
-
-function listItem(name) {
-  return `<li 
-              id="${iconId(name)}" 
-              class="btn btn-outline-dark"
-              role='button'>
-            <i class="fas fa-${name} fa-fw"></i>
-          </li>`;
-}
-
-function iconId(name) {
-  return `icon_${name.replaceAll('-', '_')}`;
-}
-
-function iconFromId(id) {
-  const m = id.match(/^icon_([\w+_]+)/);
-  if(m && m[1]) { return m[1].replaceAll('_','-') };
-}
-
-function picked(event) {
-  const icon = iconFromId(event.currentTarget.id);
-  updateIcon(icon);
-  showAllIcons();
-}
-
-function updateIcon(icon) {
-  $(`#${ICON_SHOW_ID}`).attr("class", `fas fa-${icon} fa-fw app-icon`);
-  $(`#${ICON_SELECT_ID}`).val(`fas://${icon}`);
-}
-
-function populateList() {
-  const list = $("#icon_picker_list");
-  if(list.length == 0 || ALL_ICONS.length == 0) { return; }
-
-  const listContent = ALL_ICONS.map(name => {
-    return listItem(name);
-  }).join('');
-  list.html(listContent);
-
-  ALL_ICONS.forEach(name => {
-    $(`#${iconId(name)}`).on('click', (event) => { picked(event)});
-  });
-};
+jQuery(function (){
```

Comment:
>  Refactored functions into closure to remove them from global scope.

Why make this change? This change below is still/also `document.ready` - so we've put everything in a `document.ready` closure that also has itself a document.ready closure. 

```javascript
$(function() {
```

I guess I don't understand the reasoning for the change. I kinda liked the pattern of having a bunch of functional code at the global scope and a single small `document.ready` closure.

Reply:
Looking at the code, the second `document.ready` at the end of the closure is not actually needed. It could be replaced with just the calls to the main functions:

```javascript
  populateList();
  addSearch();
  addResetForm();
```

As for adding all the utility functions within the closure instead of the global scope is just to make them private to the icon_picker.

If you prefer the previous approach, I will revert the changes.

Reply:
> As for adding all the utility functions within the closure instead of the global scope is just to make them private to the icon_picker.

You're right that they're now public functions that could collide with others potentially.  I guess I assumed they wouldn't be global.

> If you prefer the previous approach, I will revert the changes.

Yea please - I feel like a better strategy on encapsulation is correcting the import/export structures instead of writing basically every file as a script that executes.

Reply:
I reverted the changes - ready for review